prov/efa/src/efa_cq.c

```diff
@@ -69,8 +69,8 @@ ssize_t efa_cq_readerr(struct fid_cq *cq_fid, struct fi_cq_err_entry *entry,
 
 	entry->op_context = (void *)(uintptr_t)cq->ibv_cq_ex->wr_id;
 	entry->flags = efa_cq_opcode_to_fi_flags(ibv_wc_read_opcode(cq->ibv_cq_ex));
-	entry->err = EIO;
-	entry->prov_errno = cq->ibv_cq_ex->status;
+	entry->err = FI_EIO;
+	entry->prov_errno = ibv_wc_read_vendor_err(cq->ibv_cq_ex);
 	EFA_WARN(FI_LOG_CQ, "Work completion status: %s\n", ibv_wc_status_str(cq->ibv_cq_ex->status));
```

Comment:
I feel like this print may need to change based on whatever reasoning you're preferring to use ibv_wc_read_vendor_err instead of using the status enum. Or do they return the same result for EFA?

Reply:
Good point. I simply overlooked this.